### DISCUSSION TITLE
SH1122 I2C

### DISCUSSION THREAD

**Discussion by SgtKalamari** [olikraus/u8g2#1852] [Q&A] [answered]
I got the following display with an SH1122 I2C driver and i need help running it.

http://www.wide.hk/index.php?route=product/product&path=24&product_id=56

I can only find SPI in the u8g2 library but not for i2c. What am i missing?
I'm using arduino pro micro for testing

**Comment by olikraus** [comment]
Did you check https://github.com/olikraus/u8g2/wiki/u8g2setupcpp ?

**Reply by SgtKalamari** [reply]
I did but i can't really make much sense of it since i'm fairly new to Arduino.

`U8G2_SH1122_256X64_1_HW_I2C(rotation, [reset [, clock, data]]) [page buffer, size = 256 bytes]` i probably need this right?
if so how do i add it to the code?

**Comment by olikraus** [comment, marked as answer]
You could look for a HW_I2C line in one of the examples. Uncomment this line and replace the constructor name with U8G2_SH1122_256X64_1_HW_I2C

So, for example this line:
https://github.com/olikraus/u8g2/blob/7014e1d9a69883214b9bbcd64ccf0f7fd8af7964/sys/arduino/u8g2_page_buffer/HelloWorld/HelloWorld.ino#L84
becomes
```
U8G2_SH1122_256X64_1_HW_I2C u8g2(U8G2_R0, U8X8_PIN_NONE);
```

Also remember to activate 16 bit mode for your 256x64 display:
https://github.com/olikraus/u8g2/blob/7014e1d9a69883214b9bbcd64ccf0f7fd8af7964/doc/faq.txt#L163-L169

**Reply by SgtKalamari** [reply]
That worked! Thank you very much!!
